src/widgets/survey/model/hooks/useFlowState.ts

```diff
@@ -17,20 +23,61 @@ export function useFlowState({ appletId, eventId, flowId }: UseFlowStateArgs) {
     });
 
   const step = flowStorageRecord?.step ?? 0;
-  const pipeline = flowStorageRecord?.pipeline ?? [];
+  const pipeline: FlowPipelineItem[] = flowStorageRecord?.pipeline ?? [];
 
   const [isTimerElapsed, setIsTimerElapsed] = useState(false);
 
   const isLastStep = pipeline && step === pipeline.length - 1;
 
-  function next() {
+  const hasSummaryStep = pipeline && pipeline.some(x => x.type === 'Summary');
+
+  const flowSummaryData: FlowSummaryData = (flowStorageRecord?.context?.[
+    SummaryDataKey
+  ] ?? {}) as FlowSummaryData;
+
+  function next(activitySummary: ActivitySummaryData | null) {
```

Comment:
Please refactor all action functions into `useFlowActions` hook.
In `useFlowActions`, please use `getCurrentFlowStorageRecord` function to operate with the current state of a flow.

Reply:
Also, please implement `getCurrentFlowStorageRecord` function inside `useFlowStorageRecord`